core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -713,6 +717,15 @@ void assertValidUserFields(UserDto userResource, boolean nickNameIsOptional)
 		}
 
 		assertValidMobileNumber(userResource.getMobileNumber());
+
+		if (userPurpose == UserPurpose.BUDDY)
+		{
+			if (StringUtils.isBlank(userResource.getEmailAddress()))
+			{
+				throw InvalidDataException.blankEmailAddress();
+			}
+			assertValidEmailAddress(userResource.getEmailAddress());
```

Comment:
Asserting that it is valid can also be done on `UserPurpose.USER` when the e-mail address is not blank.